### PR TITLE
doc: for-expr partial evaluation.

### PR DESCRIPTION
**Pull request by i4ki**

### CODE REVIEW

File: cli/code-generation/generate-hcl.md [hunk 1]
```diff
@@ -436,6 +436,79 @@ generate_hcl "main.tf" {
...
+
+values = [
+  var.module1,
+  var.module2,
+]
+```
+
+### Partial evaluation
+
+If any part of the `for` expression contains unknown variables, then the loop construct is kept in the generated code but Terramate variables are evaluated. See example below:
```
Comment (AnnuCode) [inline cli/code-generation/generate-hcl.md:495]:
```suggestion
If any part of the 'for' expression includes unknown variables, the loop construct stays in the generated code, while Terramate variables are evaluated. See example below:
```

Comment (AnnuCode) [reply cli/code-generation/generate-hcl.md:495]: sorry, used single quotes by mistake. It will be `for`

Comment (AnnuCode) [reply cli/code-generation/generate-hcl.md:495]:
```suggestion
If any part of the 'for' expression contains unknown variables, the loop construct stays in the generated code, while Terramate variables are evaluated. See example below:
```

**Review by soerenmartius — APPROVED** [review]
lgtm

**Pull request merged**